src/systems/physics/Physics.cc

```diff
@@ -2643,6 +2653,131 @@ void PhysicsPrivate::UpdatePhysics(EntityComponentManager &_ecm)
+                .EntityCast<WorldVelocityCommandFeatureList>(_entity);
+        if (!worldLinearVelFeature)
+        {
+          static bool informed{false};
+          if (!informed)
+          {
+            gzdbg << "Attempting to set link linear velocity, but the "
+                  << "physics engine doesn't support velocity commands. "
+                  << "Velocity won't be set."
+                  << std::endl;
+            informed = true;
+          }
+          return true;
+        }
+
+        // Linear velocity in world frame
+        math::Vector3d worldLinearVel = _worldlinearvelocityreset->Data();
+
+        if(!linearVelocityResetFlag)
+        {
+          worldLinearVelFeature->SetWorldLinearVelocity(
+                                 math::eigen3::convert(worldLinearVel));
+          this->linearVelocityResetFlag = true;
+        }
+
+        return true;
+      });
+
+   // Set initial link angular velocity
+   _ecm.Each<components::Link, components::WorldAngularVelocityReset>(
+      [&](const Entity &_entity, const components::Link *,
+          const components::WorldAngularVelocityReset
+          *_worldangularvelocityreset)
+      {
+        if (!this->entityLinkMap.HasEntity(_entity))
+        {
+          gzwarn << "Failed to find link [" << _entity
+                 << "]." << std::endl;
+          return true;
+        }
+
+        auto linkPtrPhys = this->entityLinkMap.Get(_entity);
+        if (nullptr == linkPtrPhys)
+          return true;
+
+        auto freeGroup = linkPtrPhys->FindFreeGroup();
+        if (!freeGroup)
+          return true;
+
+        auto rootLinkPtr = freeGroup->RootLink();
+        if(rootLinkPtr != linkPtrPhys)
+        {
+          gzwarn << "Attempting to set angular velocity for link [ " << _entity
+                 << " ] which is not root link of the FreeGroup."
+                 << "Velocity won't be set." << std::endl;
+
+          return true;
+        }
+
+        this->entityFreeGroupMap.AddEntity(_entity, freeGroup);
+
+        auto worldAngularVelFeature =
+            this->entityFreeGroupMap
+                .EntityCast<WorldVelocityCommandFeatureList>(_entity);
+
+        if (!worldAngularVelFeature)
+        {
+          static bool informed{false};
+          if (!informed)
+          {
+            gzdbg << "Attempting to set link angular velocity, but the "
+                  << "physics engine doesn't support velocity commands. "
+                  << "Velocity won't be set."
+                  << std::endl;
+            informed = true;
+          }
+          return true;
+        }
+        // Angular velocity in world frame
+        math::Vector3d worldAngularVel = _worldangularvelocityreset->Data();
+
+        if(!angularVelocityResetFlag)
+        {
+          worldAngularVelFeature->SetWorldAngularVelocity(
+              math::eigen3::convert(worldAngularVel));
+          this->angularVelocityResetFlag = true;
+        }
```

Comment:
I see that this is your logic for ensuring that the VelocityReset components are only acted on a single time, but I recommend following the method used for the `JointVelocityReset` component, which is to delete the component after it has been acted on:

* https://github.com/gazebosim/gz-sim/blob/006602328bf55650fb0a14d9b1a17139eec6b54e/src/systems/physics/Physics.cc#L3703-L3714